File: php/examples/promotions/v1beta/ListPromotionsSample.php
<?php
/**
 * Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */


require_once __DIR__ . '/../../../vendor/autoload.php';
require_once __DIR__ . '/../../Authentication/Authentication.php';
require_once __DIR__ . '/../../Authentication/Config.php';
// [START merchantapi_list_promotions]
use Google\ApiCore\ApiException;
use Google\Shopping\Merchant\Promotions\V1beta\ListPromotionsRequest;
use Google\Shopping\Merchant\Promotions\V1beta\Promotion;
use Google\Shopping\Merchant\Promotions\V1beta\Client\PromotionsServiceClient;

/**
 * This class demonstrates how to list promotions.
 */
class ListPromotions
{

    /**
     * Lists promotions for the given account.
     *
     * @param array $config
     *      The configuration data used for authentication and getting the account ID.
     * @return void
     */
    public static function listPromotionsSample($config): void
    {

        // Gets the OAuth credentials to make the request.
        $credentials = Authentication::useServiceAccountOrTokenFile();

        // Creates options config containing credentials for the client to use.
        $options = ['credentials' => $credentials];

        // Creates a client.
        $promotionsServiceClient = new PromotionsServiceClient($options);

        try {
            // Creates the request.
            $request = new ListPromotionsRequest([
                'parent' => sprintf('accounts/%s', $config['accountId']),
            ]);

            print "Sending list promotions request:\n";

            // Makes the request.
            $response = $promotionsServiceClient->listPromotions($request);

            $count = 0;

            // Iterates over all promotions returned in the response.
            foreach ($response->iterateAllElements() as $promotion) {
                print_r($promotion);
                $count++;
            }
            printf("The following count of promotions were returned: %s\n", $count);
        } catch (ApiException $e) {
            printf("Failed to list promotions.\n");
            print $e->getMessage();
        }
    }


    /**
     * Helper to execute the sample.
     *
     * @return void
     */
    public function callSample(): void
    {
        $config = Config::generateConfig();

        // Makes the call to list promotions.
        self::listPromotionsSample($config);
    }
}


// Run the script
$sample = new ListPromotions();
$sample->callSample();
// [END merchantapi_list_promotions]

File: python/examples/promotions/v1/list_promotions_sample.py
# -*- coding: utf-8 -*-
# Copyright 2024 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
"""A module for listing Promotions."""

# [START merchantapi_list_promotions]
from examples.authentication import configuration
from examples.authentication import generate_user_credentials
from google.shopping.merchant_promotions_v1 import ListPromotionsRequest
from google.shopping.merchant_promotions_v1 import PromotionsServiceClient

_ACCOUNT = configuration.Configuration().read_merchant_info()
_PARENT = f"accounts/{_ACCOUNT}"


def list_promotions():
  """Lists promotions for the given account."""

  # Gets OAuth Credentials.
  credentials = generate_user_credentials.main()

  # Creates a client.
  client = PromotionsServiceClient(credentials=credentials)

  # Creates the request.
  request = ListPromotionsRequest(parent=_PARENT)

  # Makes the request and prints the results.
  try:
    print("Sending list promotions request:")
    response = client.list_promotions(request=request)

    count = 0

    # Iterates over all returned promotions and prints them.
    for promotion in response.promotions:
      print(promotion)
      count += 1
    print(f"The following count of promotions were returned: {count}")

  except RuntimeError as e:
    print("Failed to list promotions.")
    print(e)


if __name__ == "__main__":
  list_promotions()

# [END merchantapi_list_promotions]
